### ISSUE TITLE
Saros/I does not save open editors before session negotiation on the host side

### ISSUE THREAD

**Issue by tobous** [saros-project/saros#251]
Saros/I does not save the open editors on the host side belonging to the shared module before starting the project negotiation. This leads to inconsistencies when the editor content for one of the shared files does not match the content of the file on disk as the content of the file on disk is shared.

#### Steps to reproduce
To consistently reproduce this behavior, you will first have to disable IntelliJ automatic saving mechanisms. To do so, go into the Intellij settings. These, choose the tab **Appearance & Behavior** > **System settings** and un-select the following options:
- *Save on frame deactivation*
- *Synchronize files on frame or editor tab activation*
- *Save files automatically if application is idle for X sec.*

As the initial state, Alice and Bob both have a file A with identical content.

1. Alice changes the file A *without* saving it.
2. Alice and Bob start a session sharing the file A.

This will lead to Bob starting the session with the old content for file A, resulting in an inconsistent state.

**Comment by srossbach** [comment]
The Eclipse logic can be found here: https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/wizards/AddProjectToSessionWizard.java#L173

I think most parts have to be put into the core / IPN

**Comment by tobous** [comment]
As far as I can tell, this is already done by the core in AbstractOutgoingProjectNegotiation:
https://github.com/saros-project/saros/blob/21f19b481e225d8d6cbc20d2be9f91f01947761f/de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractOutgoingProjectNegotiation.java#L338
`EditorManager.save(IProject)` just did not work correctly in Intellij as it way relying on the internal list of open editors that was not updated correctly. This issue should be fixed with the PR #254.

Maybe the same issue also applies to Saros/E (which is why additional logic was implemented)?

Or am I misunderstanding your comment?

**Comment by srossbach** [comment]
I missed the "on the host side" part 🤦‍♂️ 
I thought you were talking about the client side.
My reply is basically a response to your solution you presented in #265

**Comment by tobous** [comment]
For the issue of moving the logic for saving editors on the client side during project negotiation, I already created the issue #232.
